src/Subscriber/EventSubscriber.php

```diff
@@ -33,46 +58,65 @@ public static function getSubscribedEvents(): array
 
     public function pushOrder(EntityWrittenEvent $event): void
     {
-        if (!isset($event->getIds()[0])) {
+        if (! isset($event->getIds()[0])) {
             return;
         }
 
-        $id = $event->getIds()[0];
-
+        $debug = false;
         try {
+            $order = $this->orderRepository->search(new Criteria([$event->getIds()[0]]), $event->getContext())->first();
+            if (! $order instanceof OrderEntity) {
+                return;
+            }
+
+            $salesChannelId = $order->getSalesChannelId();
+
+            $disabled = $this->configService->getBool($this->buildConfigKey('disabled'), $salesChannelId);
+            if ($disabled) {
+                return;
+            }
```

Comment:
Deze code zou ik in een aparte try/catch zetten, want hier kan nooit een `IncompleteConfigurationException` of `RequestFailedException` uit komen. Je hebt dan ook in de beschikking over het order id en salesChannelId als je die excepties afhandeld.